orchestra/todos/views.py

```diff
@@ -0,0 +1,89 @@
+from rest_framework import generics
+from rest_framework import permissions
+
+from orchestra.models import Task
+from orchestra.models import Todo
+from orchestra.models import Worker
+from orchestra.todos.serializers import TodoSerializer
+from orchestra.utils.notifications import message_experts_slack_group
+
+
+class IsAssociatedWithTodosProject(permissions.BasePermission):
+    """
+    Ensures that a user's worker is accoiated with the todo's project.
+    """
+
+    def has_object_permission(self, request, view, todo):
+        worker = Worker.objects.get(user=request.user)
+        project = todo.task.project
+        return worker.assignments.filter(task__project=project).exists()
+
+
+class IsAssociatedWithProject(permissions.BasePermission):
+    """
+    Ensures that a user's worker is associated with the request's
+    `project`.
+
+    """
+
+    def has_permission(self, request, view):
+        worker = Worker.objects.get(user=request.user)
+        if request.method == 'GET':
+            # List calls have a project ID
+            project_id = request.query_params.get('project')
+            return worker.assignments.filter(task__project=project_id).exists()
+        elif request.method == 'POST':
+            # Create calls have a task ID
+            task_id = request.data.get('task')
+            project = Task.objects.get(id=task_id).project
+            return worker.assignments.filter(task__project=project).exists()
+        return False
+
+
+class TodoList(generics.ListCreateAPIView):
```

Comment:
No good viewsets to let you combine the list and detail views?

Reply:
Not that I could find :P